### PR TITLE
Localization langchain prompt （zh_CN and agents)

### PR DESCRIPTION
**Pull request by lingfengchencn**
# Localization langchain prompt （now is zh_CN and agents)
Create Localization prompt for  zh_CN angets/* and will add more language and whole langchain prompt after ....

all PROMPT translated by gpt-3.5-turbo。

Fixes # (issue)
#5075

## Who can review?

Community members can review the PR once tests pass. Tag maintainers/contributors who might be interested:

  Agents / Tools / Toolkits
   @vowelparrot

### CODE REVIEW

File: langchain/utilities/locale.py [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+    """
+        define translation function
+        lang: default en_US
+        locales: default ./locale
+    """
+    domain = domain or 'langchain'
+    lang = lang or os.environ.get('LANG', 'en_US')
+    locales = locales or os.environ.get('LOCALE_PATH', 'langchain/locale')
+
+    # 加载语言包
```
Comment (Undertone0809) [inline langchain/utilities/locale.py:15]: Maybe you should change it to English.

**Comment by Undertone0809** [comment]
I think it's a good idea. But can we provide more llm support beyond gpt-3.5 turbo, or use the llm that users are using as a translation tool?

**Comment by lingfengchencn** [comment]
> I think it's a good idea. But can we provide more llm support beyond gpt-3.5 turbo, or use the llm that users are using as a translation tool?

In the beginning, I planned to just translate all the prompts into multiple languages.
Then, we need to optimize the prompt according to the different expressions of each language, so as to better adapt to the semantic context of different languages.

**Comment by Undertone0809** [comment]
> > I think it's a good idea. But can we provide more llm support beyond gpt-3.5 turbo, or use the llm that users are using as a translation tool?
> 
> In the beginning, I planned to just translate all the prompts into multiple languages.
> Then, we need to optimize the prompt according to the different expressions of each language, so as to better adapt to the semantic context of different languages.
> 
> 

In fact, I would say is it too restrictive to just use GPT 3.5 for translation? Maybe we should support arbitrary to perform translation tasks.

However, there is a problem with this. Different LLMs have different compatibility with different languages, and some are even compatible with only a few languages, so this may cause some problems. But in any case, I think it should be left to the developer to choose which LLM to use as a translation tool.

This code is so much that I didn't use it to study it carefully. By the way, where is the main code fragment translated with GPT 3.5?

**Comment by lingfengchencn** [comment]
sorry for my bad description. I have not finished all translate now,only agents.
# for codes
in langchain, all prompt template use English.
example:
```python
SYSTEM_MESSAGE_PREFIX = """Answer the following questions as best you can. You have access to the following tools:"""
```
I just use gpt-3.5-turbo to translate words ,then write down,after that  when we use English language the SYSTEM_MESSAGE_PREFIX prompt is 
"Answer the following questions as best you can. You have access to the following tools:"
and when use Chinese,the SYSTEM_MESSAGE_PREFIX promo is 
"请尽力回答以下问题。您有以下工具可供使用："

I use gpt-3.5-turbo to translate  "Answer the following questions as best you can. You have access to the following tools:" to "请尽力回答以下问题。您有以下工具可供使用：" .
I don't use llm model auto transalte the language.

So, where is the main code, nope. I use gettext ,when running ,gettext will get  LANGUAGE 、 LC_ALL 、 LC_MESSAGES and LANG  is value and get the translate word from  `locate/{language}/....`

# llm solution
you give me a new idea, we can use LLM to translate the prompt , now it has some issues when we need to parse keys.

such as :
```
"Follow this format:\n"
"\n"
"Question: input question to answer\n"
"Thought: consider previous and subsequent steps\n"
"Action:\n"
"```\n"
```
will translate to  `Chinese` :
```
请按照以下格式进行操作：

问题：输入需要回答的问题

思考：考虑前后步骤

操作：
```
or ` Spanish`
```
Siga este formato:

Pregunta: introduzca la pregunta a responder

Pensamiento: considere los pasos anteriores y posteriores

Acción:

```


and this wil raise exception...
```
regex = r"Action: (.*?)[\n]*Action Input: (.*)"
```

**Comment by leo-gan** [comment]
@lingfengchencn Hi , could you, please, resolve the merging issues and address the last comments (if needed)? After that, ping me and I push this PR for the review. Thanks!

**Comment by efriis** [comment]
Closing because the PR wouldn't line up with the current directory structure of the library (would need to be in /libs/langchain/langchain instead of /langchain). Feel free to reopen against the current head if it's still relevant!

**Pull request closed without merging**